parsers/thueringen.py

```diff
@@ -0,0 +1,148 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup as parse
+import re
+from utils import Parser
+from datetime import datetime, timedelta
+from pyopenmensa.feed import LazyBuilder
+
+amount_regex = re.compile('\d{1,2},\d{1,2}')
+employees_fee_default = -1
+guests_fee_default = -1
+
+
+def parse_start_date(document):
+	week_start_end_date_regex = re.compile('\d{1,2}\.\d{1,2}\.\d{4}')
+	calendar_week_regex = re.compile('\d{4}-\d{1,2}$')
+
+	option = document.find("option", value=calendar_week_regex, selected=True)
+
+	start_date_str = week_start_end_date_regex.search(option.text)
+
+	return datetime.strptime(start_date_str.group(), '%d.%m.%Y')
+
+
+def parse_fees(document):
+	fees_regex = re.compile('Bedienstete.*')
+
+	fees_p = document.find_all('p', {'class': 'MsoNormal'})
+
+	for fee_candidate in fees_p:
+		fee_strings = fees_regex.findall(fee_candidate.text)
+		for amount_candidate in fee_strings:
+			amount_strings = amount_regex.findall(amount_candidate)
+
+			if len(amount_strings) != 2:
+				continue
+
+			employees_fee = float(amount_strings[0].replace(',', '.'))
+			guests_fee = float(amount_strings[1].replace(',', '.'))
+			return employees_fee, guests_fee
+
+	return employees_fee_default, guests_fee_default
+
+
+def parse_meals(day):
+	ingredients_regex = re.compile('Inhalt:.*')
+
+	meals = []
+	meals_t_rows = day.find_all('tr')
+
+	for meal_data in meals_t_rows:
+		meal_t_datas = meal_data.find_all('td')
+		if len(meal_t_datas) is not 3:
+			continue
+
+		notes = [meal_t_datas[0].text]
+
+		ingredients = ingredients_regex.findall(meal_t_datas[1].text)
+		if len(ingredients) > 0:
+			notes.append(ingredients[0].strip())
+
+		main_dish = ingredients_regex.sub('', meal_t_datas[1].text).strip()
+
+		students_fee_string = amount_regex.findall(meal_t_datas[2].text)
+		if len(students_fee_string) != 1:
+			continue
+
+		students_fee = float(students_fee_string[0].replace(',', '.'))
+		costs = {'student': students_fee}
+
+		meals.append((main_dish, notes, costs))
+
+	return meals
+
+
+def parse_url(url, today=False):
+	days_regex = re.compile('day_\d$')
+
+	canteen = LazyBuilder()
+
+	content = urlopen(url).read()
+	document = parse(content, 'lxml')
```

Comment:
You are only parsing the current week currently, which means that the full fead does not contain the meals for Monday on Sunday, but last Friday instead, which is not very useful.

To get the next week, you have to send a request with the `selWeek=2018-25` parameter, where 25 is the calender week. It seems you can also get a list view using `selView=liste` which contains the explicit date, which might be easier to parse that the tabbed view.

The website uses POST-requests, but it seems GET also works: https://www.stw-thueringen.de/deutsch/mensen/einrichtungen/ilmenau/mensa-ehrenberg.html?selWeek=2018-25&selView=liste